extra/discover-host

```diff
@@ -76,15 +76,21 @@ json["discovery_version"] = version
 unless preserve_interfaces
   json["interfaces"] = interfaces.map{|i| i.first}.join(',')
   interfaces.each do |iface|
-    name, subnet, ipo, mac = iface
+    name, subnet, mac, explicit_ip = iface
     mac ||= (["52"] + 5.times.map { '%02x' % rand(0..255) }).join(':')
-    ipo ||= rand(1..253)
-    ip = "192.168.#{subnet}.#{ipo}"
     json["macaddress_#{name}"] = mac
-    json["ipaddress_#{name}"] = ip
+    subnet_woprefix, prefix = subnet.split('/')
+    if subnet.include?('.')
+      random_ip = IPAddr.new(subnet, Socket::AF_INET) | IPAddr.new(rand(2**prefix.to_i),Socket::AF_INET)
+      fact_name = "ipaddress"
+    else
+      random_ip = IPAddr.new(subnet, Socket::AF_INET6) | IPAddr.new(rand(2**prefix.to_i),Socket::AF_INET6)
+      fact_name = "ipaddress6"
+    end
+    json["#{fact_name}_#{name}"] = explicit_ip ? explicit_ip : random_ip
```

Comment:
How about generating a random ip on demand when it is actually needed?

Reply:
Good idea, but this is a helper script we only use when generating fake hosts during development. Do you insist? :-)

Reply:
Oh, I missed it is a development only thing, guess I can let it slide then :)